src/amber/router/parser.cr

```diff
@@ -0,0 +1,48 @@
+require "./parsers/**"
+
+module Amber::Router
+  module Parser
+    TYPE_EXT_REGEX   = Amber::Support::MimeTypes::TYPE_EXT_REGEX
+    URL_ENCODED_FORM = "application/x-www-form-urlencoded"
+    MULTIPART_FORM   = "multipart/form-data"
+    APPLICATION_JSON = "application/json"
+    @parsed = false
+
+    def params
+      return @params if @parsed
+      parse_url_params
+      form_data
+      json
+      multipart
```

Comment:
The order here is really delicate, and I think it might be off. Does this meansthat url parameters, which are defined in a `routes.cr` file will get overwritten by form data, json, and multipart data sources? That feels like an attack surface to me.

Reply:
Why is @params and instance variable. Seems sort of weird since it's defined outside of this scope.